rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/RxAudioPlayer.java

```diff
@@ -131,11 +135,75 @@ public void call(final SingleSubscriber<? super Boolean> singleSubscriber) {
                 }
             });
         } else {
-            return Single.error(new IllegalArgumentException(""));
+            return Observable.error(new IllegalArgumentException(""));
+        }
+    }
+
+    /**
+     * prepare audio from local file. should be scheduled in IO thread.
+     */
+    public Observable<Boolean> prepare(@NonNull final PlayConfig config) {
```

Comment:
after preparing `onNext` will be called and we can call `resume` to play the audio. and when the audio is finished `complete` will be called.

     mRxAudioPlayer.prepare(playConfig)
                .subscribeOn(Schedulers.io())
                .subscribe(new Action1<Boolean>() {
                    @Override
                    public void call(Boolean aBoolean) {//prepared
                        mRxAudioPlayer.resume();// play
                    }
                }, new Action1<Throwable>() {
                    @Override
                    public void call(Throwable throwable) {//Error

                    }
                }, new Action0() {
                    @Override
                    public void call() {
                        //end of audio
                    }

                    );